packages/main/src/plugin/kubernetes/kubernetes-port-forward-connection.ts

```diff
@@ -0,0 +1,403 @@
+
+      clearTimeout(timeoutId);
+
+      if (response.ok) {
+        resolve(disposable);
+      } else {
+        disposable.dispose();
+        reject(new Error(`Host is not reachable, received status: ${response.status}`));
+      }
+    } catch (error) {
+      disposable.dispose();
+      reject(new Error(`Host is not reachable: ${(error as Error).message}`));
+    }
+  }
+
+  /**
+   * Handles the server error event.
+   * @param error - The error instance.
+   * @param reject - The reject function of the promise.
+   * @param forwardSetup - The forwarding setup information.
+   * @throws If the port is already in use, the operation requires administrative privileges, or the server fails to initialize.
+   */
+  protected onServerError(
+    error: NodeJS.ErrnoException,
+    reject: (reason?: Error) => void,
+    forwardSetup: ForwardingSetup,
+  ): void {
+    if (error.code === 'EADDRINUSE') {
+      reject(new Error(`Port ${forwardSetup.forward.localPort} is already in use.`));
+    } else if (error.code === 'EACCES') {
+      reject(new Error('Operation requires administrative privileges.'));
+    } else {
+      reject(new Error(`Failed to initialize port forwarding server: ${error.message}`));
+    }
+  }
+
+  /**
+   * Retrieves the Kubernetes resource for the specified workload.
+   * @param kind - The kind of the workload.
+   * @param name - The name of the workload.
+   * @param namespace - The namespace of the workload.
+   * @returns The Kubernetes resource.
+   * @throws If the workload kind is not supported.
+   */
+  protected async getWorkloadResource(
+    kind: WorkloadKind,
+    name: string,
+    namespace: string,
+  ): Promise<V1Pod | V1Deployment | V1Service> {
+    if (kind === WorkloadKind.POD) {
+      return this.getPod(name, namespace);
+    } else if (kind === WorkloadKind.DEPLOYMENT) {
+      return this.getDeployment(name, namespace);
+    } else if (kind === WorkloadKind.SERVICE) {
+      return this.getService(name, namespace);
+    }
+
+    throw new Error(`Workload kind '${kind}' currently not supported.`);
+  }
+
+  /**
+   * Retrieves a pod by name and namespace.
+   * @param name - The name of the pod.
+   * @param namespace - The namespace of the pod.
+   * @returns The pod.
+   * @throws If the pod cannot be retrieved.
+   */
+  protected async getPod(name: string, namespace: string): Promise<V1Pod> {
+    return this.kubeClient.readNamespacedPod(name, namespace);
+  }
+
+  /**
+   * Retrieves a deployment by name and namespace.
+   * @param name - The name of the deployment.
+   * @param namespace - The namespace of the deployment.
+   * @returns The deployment.
+   * @throws If the deployment cannot be retrieved.
+   */
+  protected async getDeployment(name: string, namespace: string): Promise<V1Deployment> {
+    return this.kubeClient.readNamespacedDeployment(name, namespace);
+  }
```

Comment:
Is there really a point to port forward deployment ?

Reply:
This is implememented by kubectl so let's keep the code and decide at UI level

Reply:
Oh I was not aware, thanks for the info